models/demos/qwen/tests/test_qwen_model.py

```diff
@@ -0,0 +1,275 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+import torch
+import pytest
+from loguru import logger
+import os
+import ttnn
+from models.demos.qwen.tt.qwen_common import (
+    precompute_freqs,
+    get_single_rot_mat,
+    sample,
+    HostEmbedding,
+)
+from models.demos.qwen.tt.qwen_model import TtTransformer
+from models.demos.qwen.tt.model_config import TtModelArgs
+from models.demos.qwen.reference.model import Transformer
+from models.demos.qwen.reference.tokenizer import Tokenizer
+from models.utility_functions import (
+    comp_pcc,
+    comp_allclose,
+)
+from models.utility_functions import skip_for_grayskull
+
+
+@torch.no_grad()
+@skip_for_grayskull("Requires wormhole_b0 to run")
+@pytest.mark.timeout(900)
+@pytest.mark.models_performance_bare_metal
+@pytest.mark.parametrize(
+    "weights, layers",
+    [
+        ("random", 1),
+        ("instruct", None),
+    ],
+    ids=["quick", "full"],
+)
+@pytest.mark.parametrize(
+    "mesh_device",
+    [
+        {"N150": (1, 1), "N300": (1, 2), "T3K": (1, 8), "TG": (8, 4)}.get(
+            os.environ.get("FAKE_DEVICE"), len(ttnn.get_device_ids())
+        )
+    ],
+    indirect=True,
+)
+def test_qwen_model_inference(mesh_device, weights, layers, use_program_cache, reset_seeds, ensure_gc):
+    run_ref_pt = True  # Flag to run reference PyTorch model and compare PCC
+    cache_pcc = layers == 1  # Flag to measure KV cache PCC. Avoid running for all layers to speed up test time.
+
+    dtype = ttnn.bfloat8_b
+
+    mesh_device.enable_async(True)
+
+    # This sets the minimum PCC for each iteration
+    pcc = 0.79
```

Comment:
do we need to use tight pcc (10 decimals) to prevent regressions since we are not doing token matching?

Reply:
updated to a tighter pcc for last iteration